sktime/transformations/series/window_summarizer.py

```diff
@@ -0,0 +1,357 @@
+            * "max",
+            * "corr",
+            * "cov",
+            * "skew",
+            * "sem"
+         or function definition. See for the native functions also
+         https://pandas.pydata.org/docs/reference/window.html.
+    window: list of integers
+        Contains values for window shift and window length.
+    """
+    win = window[0]
+    shift = window[1] + 1
+
+    if win_summarizer in pd_rolling:
+        if isinstance(Z.index, pd.MultiIndex):
+            feat = getattr(Z.shift(shift).rolling(win), win_summarizer)()
+        else:
+            feat = pd.DataFrame(Z).apply(
+                lambda x: getattr(x.shift(shift).rolling(win), win_summarizer)()
+            )
+    else:
+        feat = Z.shift(shift)
+        if isinstance(Z.index, pd.MultiIndex) and callable(win_summarizer):
+            feat = feat.rolling(win).apply(win_summarizer, raw=True)
+        elif not isinstance(Z.index, pd.MultiIndex) and callable(win_summarizer):
+            feat = feat.apply(lambda x: x.rolling(win).apply(win_summarizer, raw=True))
+
+    if isinstance(feat, pd.Series):
+        feat = pd.DataFrame(feat)
+
+    feat.rename(
+        columns={
+            feat.columns[0]: name + "_" + "_".join([str(item) for item in window])
+        },
+        inplace=True,
+    )
+
+    return feat
+
+
+class LaggedWindowSummarizer(BaseTransformer):
+    """Transformer for extracting time series features.
+
+    The LaggedWindowSummarizer transforms input series to features
+    based on a provided dictionary of window summarizer, window shifts
+    and window lengths.
+
+    Parameters
+    ----------
+    Dictionary with the following arguments:
+    index : str, base string of the derived features
+            final name will also contain
+            window shift and window length.
+    win_summarizer: either str corresponding to implemented pandas function, currently
+            * "sum",
+            * "mean",
+            * "median",
+            * "std",
+            * "var",
+            * "kurt",
+            * "min",
+            * "max",
+            * "corr",
+            * "cov",
+            * "skew",
+            * "sem"
+         or function definition
+    window: list of integers
+        Contains values for window shift and window length.
+    target_cols: list of str,
+        Specifies which columns in y or X to target. If set to None will
+        target first column in X.
+
+    Examples
+    --------
+    >>> from sktime.transformations.series.window_summarizer import (
+    ...    LaggedWindowSummarizer
+    ... )
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> kwargs_variant = {
+    ...     "functions": {
+    ...     "mean": ["mean", [[1, 7], [8, 7]]],
+    ...     "covar_feature": ["cov", [[1, 28]]],
+    ...     }
+    ...  }
+    >>> transformer = LaggedWindowSummarizer(**kwargs_variant)
+    >>> y_transformed = transformer.fit_transform(y)
+    """
```

Comment:
the use within make_reduction does not yet work, since the hierarchical PR is not finished. However I included two instructive ForecastingPipeline examples